units/src/amount.rs

```diff
@@ -1155,10 +1165,27 @@ impl ops::DivAssign<u64> for Amount {
     fn div_assign(&mut self, rhs: u64) { *self = *self / rhs }
 }
 
+#[cfg(feature = "alloc")]
 impl FromStr for Amount {
     type Err = ParseError;
 
-    fn from_str(s: &str) -> Result<Self, Self::Err> { Amount::from_str_with_denomination(s) }
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let a = Amount::from_str_with_denomination(s);
+
+        match a {
+            Err(ParseError::MissingDenomination(_)) => {
+                let d = s.to_string() + " sat";
+                let w = Amount::from_str_with_denomination(&d);
```

Comment:
Don't do these string operations, call `Amount::from_str_in(Denomination::Satoshi)` instead.

Reply:
Ah cool, that's how to avoid needing alloc.